src/blockchain.erl

```
@@ -892,12 +892,18 @@ find_first_height_after(MinHeight0, #blockchain{db=DB, heights=HeightsCF}) ->
     MinHeight = max(0, MinHeight0),
     {ok, Iter} = rocksdb:iterator(DB, HeightsCF, []),
     rocksdb:iterator_move(Iter, {seek, <<(MinHeight):64/integer-unsigned-big>>}),
-    case rocksdb:iterator_move(Iter, next) of
-        {ok, <<Height:64/integer-unsigned-big>>, Hash} ->
-            {ok, Height, Hash};
-        {error, _} ->
-            {error, not_found}
-    end.
+    Result =
+        case rocksdb:iterator_move(Iter, next) of
+            {ok, <<Height:64/integer-unsigned-big>>, Hash} ->
+                {ok, Height, Hash};
+            {error, invalid_iterator} ->
+                {error, not_found};
+            {error, Reason} ->
+                lager:error("Unexpected iterator error: ~p", [Reason]),
+                {error, not_found}
+        end,
+    catch rocksdb:iterator_close(Iter),
```

thought we were trying to avoid these types of catches since they swallowed errors

---

Yeah. I hate it. I've done a bunch of rocks iteration experiments recently and stumbled on enough seemingly-nondeterministic iterator-access errors to make me nervous.

While I lean on removing these `catch`s here, I'm not sure what will happen if you try to close an iterator after an error other than `invalid_iterator` - hence the recon logging.

I wasn't able to reproduce those not-`invalid_iterator` errors.

---

@jeffgrunewald OK, I replaced all the `catch rocksdb:iterator_close`s with `try rocksdb:iterator_close catch log end`s.